<a href="https://colab.research.google.com/github/AmelieSnn/TitanicData/blob/main/EXO_Gouvernance_des_donn%C3%A9es.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>


# TP 1 Gouvernance des Données


## 1. Exercice : Evaluation et Amélioration de la Qualité des Donnnées


**• Objectif**: Evaluer les problemes de qualite des donnees et mettre
en œuvre des strategies d’amelioration.

**• Source des Donnees**: Utilisez le jeu de donnees ”Titanic” de Titanic - Machine Learning from Disaster, qui comprend des in- formations sur les passagers du Titanic, telles que l’age, la classe, le genre et le statut de survie. vous pouvez utiliser celle de ’train’.

**• Tâches :**

(a) Evaluer le jeu de donnees pour identifier les problemes de qualite, y compris les valeurs manquantes, les doublons, les incoherences et les valeurs aberrantes.

(b) Nettoyer les donn ́ees en gerant les valeurs manquantes, en supprimant les doublons et en corrigeant les incoh ́erences.

(c) Rediger un rapport detaillant les problemes de qualite identifies, les etapes suivies pour nettoyer les donnees, et des recommandations pour la gestion continue de la qualite des donnees.


**• Resultat Attendu :**

– Un jeu de donnees nettoye avec une qualite amelioree.

– Un rapport d ́ecrivant les probl`emes de qualite des donnees, les actions entreprises et une proposition pour maintenir la qualit ́e des donnees.


In [1]:
# Import des librairies pandas et numpy.
import pandas as pd
import numpy as np

In [3]:
# IMPORTATION DE DRIVE
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [4]:
# IMPORTATION DU DATASET (ET SUPPRESSION DE LA COLONNE UNNAMED : 0)
df = pd.read_csv("/content/drive/MyDrive/IASCHOOL/MASTER 2/Gouvernance des données/train.csv", index_col = 'PassengerId')

In [5]:
# Affichage du dataset
df.head(20)

Survived  Pclass  \
PassengerId                     
1                   0       3   
2                   1       1   
3                   1       3   
4                   1       1   
5                   0       3   
6                   0       3   
7                   0       1   
8                   0       3   
9                   1       3   
10                  1       2   
11                  1       3   
12                  1       1   
13                  0       3   
14                  0       3   
15                  0       3   
16                  1       2   
17                  0       3   
18                  1       2   
19                  0       3   
20                  1       3   

                                                          Name     Sex   Age  \
PassengerId                                                                    
1                                      Braund, Mr. Owen Harris    male  22.0   
2            Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0   
3                                       Heikkinen, Miss. Laina  female  26.0   
4                 Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0   
5                                     Allen, Mr. William Henry    male  35.0   
6                                             Moran, Mr. James    male   NaN   
7                                      McCarthy, Mr. Timothy J    male  54.0   
8                               Palsson, Master. Gosta Leonard    male   2.0   
9            Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0   
10                         Nasser, Mrs. Nicholas (Adele Achem)  female  14.0   
11                             Sandstrom, Miss. Marguerite Rut  female   4.0   
12                                    Bonnell, Miss. Elizabeth  female  58.0   
13                              Saundercock, Mr. William Henry    male  20.0   
14                                 Andersson, Mr. Anders Johan    male  39.0   
15                        Vestrom, Miss. Hulda Amanda Adolfina  female  14.0   
16                            Hewlett, Mrs. (Mary D Kingcome)   female  55.0   
17                                        Rice, Master. Eugene    male   2.0   
18                                Williams, Mr. Charles Eugene    male   NaN   
19           Vander Planke, Mrs. Julius (Emelia Maria Vande...  female  31.0   
20                                     Masselmani, Mrs. Fatima  female   NaN   

             SibSp  Parch            Ticket     Fare Cabin Embarked  
PassengerId                                                          
1                1      0         A/5 21171   7.2500   NaN        S  
2                1      0          PC 17599  71.2833   C85        C  
3                0      0  STON/O2. 3101282   7.9250   NaN        S  
4                1      0            113803  53.1000  C123        S  
5                0      0            373450   8.0500   NaN        S  
6                0      0            330877   8.4583   NaN        Q  
7                0      0             17463  51.8625   E46        S  
8                3      1            349909  21.0750   NaN        S  
9                0      2            347742  11.1333   NaN        S  
10               1      0            237736  30.0708   NaN        C  
11               1      1           PP 9549  16.7000    G6        S  
12               0      0            113783  26.5500  C103        S  
13               0      0         A/5. 2151   8.0500   NaN        S  
14               1      5            347082  31.2750   NaN        S  
15               0      0            350406   7.8542   NaN        S  
16               0      0            248706  16.0000   NaN        S  
17               4      1            382652  29.1250   NaN        Q  
18               0      0            244373  13.0000   NaN        S  
19               1      0            345763  18.0000   NaN        S  
20               0      0              2649   7.2250   

In [6]:
# Informations sur le dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 891 entries, 1 to 891
Data columns (total 11 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Survived  891 non-null    int64  
 1   Pclass    891 non-null    int64  
 2   Name      891 non-null    object 
 3   Sex       891 non-null    object 
 4   Age       714 non-null    float64
 5   SibSp     891 non-null    int64  
 6   Parch     891 non-null    int64  
 7   Ticket    891 non-null    object 
 8   Fare      891 non-null    float64
 9   Cabin     204 non-null    object 
 10  Embarked  889 non-null    object 
dtypes: float64(2), int64(4), object(5)
memory usage: 83.5+ KB


In [7]:
# Les différentes modalités prises par les différentes colonnes

valeurs_uniques = df.apply(lambda col: col.unique())
print(valeurs_uniques)

Survived                                               [0, 1]
Pclass                                              [3, 1, 2]
Name        [Braund, Mr. Owen Harris, Cumings, Mrs. John B...
Sex                                            [male, female]
Age         [22.0, 38.0, 26.0, 35.0, nan, 54.0, 2.0, 27.0,...
SibSp                                   [1, 0, 3, 4, 2, 5, 8]
Parch                                   [0, 1, 2, 5, 3, 4, 6]
Ticket      [A/5 21171, PC 17599, STON/O2. 3101282, 113803...
Fare        [7.25, 71.2833, 7.925, 53.1, 8.05, 8.4583, 51....
Cabin       [nan, C85, C123, E46, G6, C103, D56, A6, C23 C...
Embarked                                       [S, C, Q, nan]
dtype: object


In [8]:
# Statistiques du dataset

df.describe()

Survived      Pclass         Age       SibSp       Parch        Fare
count  891.000000  891.000000  714.000000  891.000000  891.000000  891.000000
mean     0.383838    2.308642   29.699118    0.523008    0.381594   32.204208
std      0.486592    0.836071   14.526497    1.102743    0.806057   49.693429
min      0.000000    1.000000    0.420000    0.000000    0.000000    0.000000
25%      0.000000    2.000000   20.125000    0.000000    0.000000    7.910400
50%      0.000000    3.000000   28.000000    0.000000    0.000000   14.454200
75%      1.000000    3.000000   38.000000    1.000000    0.000000   31.000000
max      1.000000    3.000000   80.000000    8.000000    6.000000  512.329200

In [9]:
# Présence de doublons
df.duplicated().sum()

0

In [10]:
# Présence de valeurs manquantes
df.isna().sum().sum()

866

## Detection de valeurs aberrantes

**Survived**

**Analyse** : La colonne Survived est binaire (0 = non survécu, 1 = survécu).

**Min/Max** : min = 0 et max = 1. Ces valeurs sont logiques car il n’y a que deux résultats possibles.

**Conclusion** : Pas de valeurs aberrantes ici, car les données correspondent aux résultats attendus (0 ou 1).

**Pclass**

**Analyse** : La colonne Pclass représente la classe du passager (1ère, 2ème, ou 3ème classe).

**Min/Max** : min = 1 et max = 3. Ces valeurs sont logiques puisqu'il n'y avait que 3 classes à bord.

**Conclusion** : Pas de valeurs aberrantes, car les valeurs sont comprises entre 1 et 3.

**Name**

**Analyse** : Cette colonne contient des chaînes de caractères (noms des passagers). Comme il s'agit d'une donnée textuelle, on ne peut pas détecter de valeurs aberrantes ici avec des statistiques descriptives classiques.

**Conclusion** : Aucune détection automatique de valeurs aberrantes n'est applicable pour les chaînes de caractères.

**Sex**

**Analyse** : La colonne Sex contient des valeurs catégorielles : "male" ou "female".

**Détection d'aberrations**: Ici, on voit qu'il n'existe pas d'autres valeurs que "male" et "female", il n'y a donc pas de valeurs aberrantes

**Conclusion** : Aucune valeur aberrante

**Age**

**Analyse** : La colonne Age représente l'âge des passagers

**Min/Max**: min = 0.42 (âge minimum) et max = 80. La valeur de min = 0.42 pourrait correspondre à un bébé de moins de 6mois, donc elle semble plausible. Le maximum (80 ans) est également plausible.

**Conclusion** : Aucune valeur aberrante